File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() == 0) return true;
        if (s.size() > t.size()) return false;
        int ptr = 0; // Ptr for the char in s
        for (auto c : t) {
            if (c == s[ptr]) {
                ptr++;
                if (ptr >= s.size()) {
                    return true;
                }
            }
        }
        
        return false;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0: return True
        if len(s) > len(t): return False
        ptr = 0
        for c in t:
            if c == s[ptr]:
                ptr += 1
                if ptr >= len(s):
                    return True
                
        return False